staging/src/k8s.io/kubectl/pkg/cmd/exec/exec.go

```diff
@@ -377,3 +369,31 @@ func (p *ExecOptions) Run() error {
 
 	return nil
 }
+
+func getDefaultContainerName(p *ExecOptions, pod *corev1.Pod) string {
```

Comment:
Maybe this can be an exported helper function, so anyone wanting to get the default container name can use this (It can be easier also to implement later on the commands, like 'GetDefaultContainerName' and if this is nil, then follow with the current behavior (--container, error, etc)